### PR TITLE
feat: Move tracing to IAVL tree substore for Fraud Proof implementation

### LINKED ISSUE #9: Move tracing to IAVL tree substore for Fraud Proof implementation
We'd like to modify the witness data in the `FraudProof` struct to use witness data generated by the IAVL tree substore in https://github.com/celestiaorg/iavl/issues/19

This enables us to use the IAVL tree substore entirely for tracing and during verification of fraud proofs, we can load this witness data into a Deep Subtree so that it performs the same underlying tree operations for a fraudulent transaction.

Relevant changes to accommodate this design change should be made in ABCI methods `GenerateFraudProof` and `VerifyFraudProof` (and methods used by them).

We should also delete any new tracing related things we added to `tracekv` and `rootmulti` store on the cosmos-sdk level since all the tracing happens in the underlying IAVL tree substore now.

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Description

We'd like to modify the witness data in the `FraudProof` struct to use witness data generated by the IAVL tree substore in https://github.com/celestiaorg/iavl/issues/19

This enables us to use the IAVL tree substore entirely for tracing and during verification of fraud proofs, we can load this witness data into a Deep Subtree so that it performs the same underlying tree operations for a fraudulent transaction.

Relevant changes to accommodate this design change should be made in ABCI methods `GenerateFraudProof` and `VerifyFraudProof` (and methods used by them).

We should also delete any new tracing related things we added to `tracekv` and `rootmulti` store on the cosmos-sdk level since all the tracing happens in the underlying IAVL tree substore now.

Closes: #9 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: baseapp/abci.go [hunk 1]
```diff
@@ -245,16 +247,19 @@ func (app *BaseApp) GenerateFraudProof(req abci.RequestGenerateFraudProof) (res
...
 	abciFraudProof := req.FraudProof
 	fraudProof := FraudProof{}
-	fraudProof.fromABCI(*abciFraudProof)
+	err := fraudProof.fromABCI(*abciFraudProof)
+	if err != nil {
+		panic(err)
+	}
 
-	// First two levels of verification
+	// Store and subtore level verification
```
Comment (tzdybal) [inline baseapp/abci.go:255]: :+1:

File: baseapp/baseapp.go [hunk 2]
```diff
@@ -884,37 +878,27 @@ func (app *BaseApp) getFraudProof(storeKeyToTraceBuf map[string]*bytes.Buffer) (
 		stateWitness := StateWitness{
 			Proof:       *proof,
 			RootHash:    rootHash,
-			WitnessData: make([]WitnessData, 0),
+			WitnessData: make([]*WitnessData, 0),
```
Comment (tzdybal) [inline baseapp/baseapp.go]: You can set capacity of this slice.

**Review by tzdybal — COMMENTED** [review]
Left few comments.

File: baseapp/baseapp.go (hunk 2 above)
Comment (Manav-Aggarwal) [reply baseapp/baseapp.go]: Addressed. Feel free to resolve

**Review by tzdybal — APPROVED** [review]

**Pull request merged**